### PR TITLE
feat: Add analysis agent for deterministic JIRA routing gate

### PR DESCRIPTION
**Pull request by yyualvin**
Introduces Agent 1 (analysis_agent.py) which sets is_open on RCA result rows based on ticket_link and ticket_resolve_datetime_gmt using a 4-hour residual window. Includes unit tests covering the business rule edge cases and adds the uv.lock file.

### CODE REVIEW

File: deploy/batch-rca-automation/scripts/analysis_agent.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        return True
+
+    resolved_at = ticket_resolve_datetime_gmt
+    if not resolved_at.tzinfo:
+        resolved_at = resolved_at.replace(tzinfo=timezone.utc)
+
+    delta = job_finished - resolved_at
+    if delta < timedelta(hours=JIRA_RESIDUAL_WINDOW_HOURS):
+        return False  # residual — ticket recently closed
+    return True  # new incident — fix was in place before this job ran
```
Comment (jerry-ng2) [inline deploy/batch-rca-automation/scripts/analysis_agent.py:55]: instead of calculating it from the time the job finished can this be done with a sql query instead. `resolved_at > NOW() - '4 hours'`. This way it can be done at time of execution of the job instead of looking at the job_finished time

File: deploy/batch-rca-automation/scripts/analysis_agent.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    group.add_argument(
+        "--lookback-hours",
+        type=int,
+        default=None,
+        help="Process jobs finished in the last N hours (e.g. 24)",
+    )
+    args = parser.parse_args(argv)
+
+    if args.since:
+        since = datetime.fromisoformat(args.since).replace(tzinfo=timezone.utc)
```
Comment (jerry-ng2) [inline deploy/batch-rca-automation/scripts/analysis_agent.py:156]: This seems repetitive these flags already appear in `query_source_db.py` and `query_historical_matches.py`.  If you need them here then create it in `utils.py` to reduce redundancy.

**Review by jerry-ng2 — CHANGES_REQUESTED** [review]
Few changes need to be made but after the e2e testing to confirm it works it should be fine.

**Comment by jerry-ng2** [comment]
Also needs to be invoked somewhere in `batch_rca_headless.sh`

**Comment by yyualvin** [comment]
> Also needs to be invoked somewhere in `batch_rca_headless.sh`

This can only be invoked after batch analysis as of now because it relies on `aap2_result` table having the fields `ticket_link` and `ticket_resolve_datetime_gmt`
